dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+        ContentletAPI conAPI = APILocator.getContentletAPI();
+        int limit = 500;
+        int offset = 0;
+        List<Contentlet> contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+            .getSystemUser(), false, limit, offset);
+        int size = contentlets.size();
+        while (size > 0) {
+            for (Contentlet contentlet : contentlets) {
+                invalidateAll(contentlet);
+            }
+            offset += limit;
+            contentlets = conAPI.findByStructure(structure, APILocator.getUserAPI()
+                .getSystemUser(), false, limit, offset);
+            size = contentlets.size();
+        }
+    }
+
+    private void removeContentletFile(Contentlet asset, Identifier identifier, boolean EDIT_MODE)
+            throws DotDataException, DotSecurityException {
+        CacheLocator.getContentletCache()
+            .remove(asset.getInode());
+
+        String folderPath = (!EDIT_MODE) ? "live" + java.io.File.separator : "working" + java.io.File.separator;
+        String velocityRootPath = VelocityUtil.getVelocityRootPath();
+        velocityRootPath += java.io.File.separator;
+
+        Set<Long> langs = new HashSet<Long>();
+        langs.add(asset.getLanguageId());
+        if (LanguageWebAPI.canApplyToAllLanguages(asset)) {
+            for (Language ll : APILocator.getLanguageAPI()
+                .getLanguages()) {
+                langs.add(ll.getId());
+            }
+        }
+        for (Long langId : langs) {
+            String filePath = folderPath + identifier.getId() + "_" + langId + "." + VelocityType.CONTENT.fileExtension;
+            java.io.File f = new java.io.File(velocityRootPath + filePath);
+            f.delete();
+            DotResourceCache vc = CacheLocator.getVeloctyResourceCache();
+            vc.remove(ResourceManager.RESOURCE_TEMPLATE + filePath);
+        }
+        List<Field> fields = asset.getContentType()
+            .fields();
+        for (Field field : fields) {
+            new FieldLoader().invalidate(field);
+        }
+    }
+
+
+
+    @Override
+    public InputStream writeObject(String id1, String id2, boolean live, String language, String filePath) {
+        try {
+
+            Identifier identifier = APILocator.getIdentifierAPI()
+                .find(id1);
+            Contentlet contentlet = null;
+            if (CacheLocator.getVeloctyResourceCache()
+                .isMiss(filePath)) {
+                if (LanguageWebAPI.canDefaultContentToDefaultLanguage()) {
+                    LanguageAPI langAPI = APILocator.getLanguageAPI();
+                    language = Long.toString(langAPI.getDefaultLanguage()
+                        .getId());
+                } else {
+                    throw new ResourceNotFoundException("Contentlet is a miss in the cache");
+                }
+            }
+
+            try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)